### PR TITLE
Fix: Added removal of subscriptions on complete

### PR DESCRIPTION
**Pull request by Soumyajit-Saha**
Previously there was only a single disposal for all the subsriptions. So, all the subscriptions were getting cancelled when the disposal bag was going out of scope and not when a completion request was received on a particular subscription. So, I added a dictionary of disposal bags with subscription 'id' as key and the disposal bag for that subscription as value. On subscription, a new disposal bag corresponding to the subscription is created and inserted in the dictionary. On completion, the disposal bag is removed from the dictionary. Now, the test for consecutive subscription of same operation is passing.

Link: https://app.clickup.com/t/866amdey9

### CODE REVIEW

**Comment by GNMoseke** [comment]
Excellent catch @Soumyajit-Saha! LGTM, I'll give @NeedleInAJayStack final say on this one

**Review by GNMoseke — APPROVED** [review]

File: Sources/GraphQLTransportWS-DataSync/Server.swift [hunk 1]
```diff
@@ -207,6 +208,8 @@ public class Server<InitPayload: Equatable & Codable> {
                 // swiftlint:disable:next force_cast
                 let stream = streamOpt as! ObservableSubscriptionEventStream
                 let observable = stream.observable
+                let disposeBag = DisposeBag()
+                disposeBags[id] = disposeBag
```
Comment (NeedleInAJayStack) [inline Sources/GraphQLTransportWS-DataSync/Server.swift:220]: nit: This isn't a bug you caused, but now that we're introducing state around the `id` values, we should probably check that the provided `id` is not already taken (that the client hasn't double-subscribed on a particular ID).

**Review by NeedleInAJayStack — COMMENTED** [review]
Nice Soumyajit! Just one comment.

File: Sources/GraphQLTransportWS-DataSync/Server.swift (hunk 1 above)
Comment (Soumyajit-Saha) [reply Sources/GraphQLTransportWS-DataSync/Server.swift:220]: Yes, it makes sense. I'll make the change.

Comment (NeedleInAJayStack) [reply Sources/GraphQLTransportWS-DataSync/Server.swift:220]: Excellent! Thanks!

**Pull request merged**